src/windows/main_window.py

```diff
@@ -1332,6 +1335,20 @@ def actionNextMarker_trigger(self, event):
             get_app().window.refreshFrameSignal.emit()
             get_app().window.propertyTableView.select_frame(frame_to_seek)
 
+    def addKeyframesToMarkers(self, data):
+        for property in data :
+            try :
+                for point in data[property]["Points"] :
+                    keyframe=(point["co"]["X"]-1)/fps_float - data["start"] + data["position"]
+                    if keyframe > start and keyframe < stop :
+                        all_marker_positions.append(keyframe)
```

Comment:
As with `fps_float`, `all_marker_positions` is undefined inside this function. You can't make references to variables from your caller.

The way I'd suggest to do this is, inside the function you should create a _list_ of the marker positions you want to add, and return that back to the caller. Then, you can add its results to the `all_marker_positions` list from the calling method. So, basically like this:

```python
# In your function -- which, again, needs to be defined BEFORE it's called — build and return a list:
# (I also renamed it to better indicate its purpose, since it can't directly add to the markers list)
def getKeyframePositions(data):
    positions = []
    for property in data:
        # (Do the work to compute a position...)
        positions.append(keyframe_time)
    return positions

# And then in either of the places where you call the function:
all_marker_positions.extend(getKeyframePositions(selected_clip.data))
```

Using `list.extend()` is crucial here, as it combines one list with the contents of another, rather than adding the additional list as a single element.
```python
# If you have a list like this:
>>> mylist = [1, 2, 3]
# the difference is:
>>> mylist.append([4, 5, 6])
# mylist = [1, 2, 3, [4, 5, 6]]
>>> mylist.extend([4, 5, 6])
# mylist = [1, 2, 3, 4, 5, 6]
```

Reply:
I was going to say that writing this as a function means you _have_ to pass in `fps_float` from the caller, but I don't think that's right. `get_app()` should work from a function as well as from a method. So it's your call whether you pass it in, or compute it from the value of `get_app().project.get("fps")` inside your function.